bin/generate_schema.py

```diff
@@ -0,0 +1,233 @@
+  manylinux-pypy_i686-image:
+    type: string
+    description: Specify alternative manylinux / musllinux container images
+  manylinux-pypy_x86_64-image:
+    type: string
+    description: Specify alternative manylinux / musllinux container images
+  manylinux-s390x-image:
+    type: string
+    description: Specify alternative manylinux / musllinux container images
+  manylinux-x86_64-image:
+    type: string
+    description: Specify alternative manylinux / musllinux container images
+  musllinux-aarch64-image:
+    type: string
+    description: Specify alternative manylinux / musllinux container images
+  musllinux-i686-image:
+    type: string
+    description: Specify alternative manylinux / musllinux container images
+  musllinux-ppc64le-image:
+    type: string
+    description: Specify alternative manylinux / musllinux container images
+  musllinux-s390x-image:
+    type: string
+    description: Specify alternative manylinux / musllinux container images
+  musllinux-x86_64-image:
+    type: string
+    description: Specify alternative manylinux / musllinux container images
+  repair-wheel-command:
+    type: string_array
+    description: Execute a shell command to repair each built wheel.
+  skip:
+    description: Choose the Python versions to skip.
+    type: string_array
+  test-command:
+    description: Execute a shell command to test each built wheel.
+    type: string_array
+  test-extras:
+    description: Install your wheel for testing using `extras_require`
+    type: string_array
+  test-requires:
+    description: Install Python dependencies before running the tests
+    type: string_array
+  test-skip:
+    description: Skip running tests on some builds.
+    type: string_array
+"""
+
+schema = yaml.safe_load(starter)
+
+string_array = yaml.safe_load(
+    """
+- type: string
+- type: array
+  items:
+    type: string
+"""
+)
+
+string_table_array = yaml.safe_load(
+    """
+- type: string
+- patternProperties:
+    .+:
+      oneOf:
+        - type: string
+        - type: array
+          items:
+            type: string
+"""
+)
+
+string_table = yaml.safe_load(
+    """
+- type: string
+- patternProperties:
+    .+:
+      - type: string
+"""
+)
+
+for key, value in schema["properties"].items():
+    value["title"] = f'CIBW_{key.replace("-", "_").upper()}'
+    match value:
+        case {"type": "string_array"}:
+            del value["type"]
+            value["oneOf"] = string_array
+        case {"type": "string_table"}:
+            del value["type"]
+            value["oneOf"] = string_table
+        case {"type": "string_table_array"}:
+            del value["type"]
+            value["oneOf"] = string_table_array
+
+overrides = yaml.safe_load(
+    """
+type: array
+description: An overrides array
+items:
+  type: object
+  required: ["select"]
+  additionalProperties: false
+  properties:
+    select: {}
+"""
+)
+
+limited = schema["properties"].copy()
+del limited["build"]
+del limited["skip"]
+
```

Comment:
Also, `test-skip` and `archs` are global. Archs cannot be overridden in an override.